Package.swift

```diff
@@ -1,4 +1,4 @@
-// swift-tools-version:5.2
```

Comment:
If this PR ends up changing the minimum supported Swift version from 5.2 to 5.5, that will be a breaking change for any consumers who support compiling with Swift < 5.5. Is the intention to ship this as part of a 2.0 release or is breaking Swift version support not considered a semver breaking change?

Reply:
Good question, @jpsim! The plan is to ship this change as part of the 1.1.0 release. My understanding is that this isn't a source breaking change, since clients using older versions of Swift will just continue to get version 1.0.3. That said, I do have the wrong platform setting below — will update that. Thanks!

Reply:
Is there a reason why we don't wrap the new stuff in `#if swift(>=5.5)`, and in turn keep support for older Swift versions?

Reply:
@ffried Not to say that you shouldn't, what are your use cases where you need to support older swift versions like 5.2?

Reply:
@rauhul I don't have any myself. I was just wondering in case this is considered a breaking change.

Reply:
Usage of older Swift versions like 5.1/5.2 is higher than you might think ([15% of this survey's respondents](https://twitter.com/daveverwer/status/1489287264043773955)). So for library authors who want to continue supporting those users, all dependencies must be compatible with all supported versions.

My original question though was more about semantic versioning and whether changing the minimum supported Swift version constitutes a breaking change.

> clients using older versions of Swift will just continue to get version 1.0.3

I could be wrong, but I don't think that's how SwiftPM works. I suspect it'll see the `// swift-tools-version:5.5` comment and fail immediately if the swift tools version is older than that.

Reply:
> Usage of older Swift versions like 5.1/5.2 is higher than you might think ([15% of this survey's respondents](https://twitter.com/daveverwer/status/1489287264043773955)).

That might be a convincing enough reason to upgrade. IMO _we_ (the community + apple package owners) should come up with some guidelines for version support. I think NIO announced they will support only the latest 2 swift versions. (Though I might be mis-remembering). 

@Lukasa what is swift-nio's police for swift version support and do you consider dropping support for a swift version a breaking change?

Reply:
We are proposing to support [the latest Swift version and the two versions prior](https://forums.swift.org/t/swiftnio-swift-version-support/53232). We do not consider dropping Swift versions to be a breaking change, because as @natecook1000 says, SwiftPM takes the tools version into account when resolving what version of a package will be used. Users silently get the last version that supported their Swift version.

Reply:
> SwiftPM takes the tools version into account when resolving what version of a package will be used. Users silently get the last version that supported their Swift version

This is a big surprise to me since it means SwiftPM needs to check out multiple versions of a dependency in order to identify which on is the last to be compatible with the current swift version. Until today I thought SwiftPM just looked at the git tags to determine which version of a dependency to resolve.

If this is the case, then I have no objections to versioning this with 1.1.0.

@rauhul it'd be great if there were guidelines for Swift version support for Swift library authors.

Reply:
@jpsim This seems like a good conversation to start on the swift forums, I'm 100% certain people have opinions... (for better or worse)

Reply:
A fun irony here is that we discussed the tools version change, but I missed that this _also_ added a `platforms` stanza, and while changing the tools version isn't breaking, changing the platforms stanza very much is.